cli/src/pcluster/validators/networking_validators.py

```diff
@@ -198,3 +198,14 @@ def _validate_all_subnets_exist(self, existing_subnets, expected_subnet_ids):
                 f"Some subnets associated to the Lambda are not present in the account: {sorted(missing_subnet_ids)}.",
                 FailureLevel.ERROR,
             )
+
+
+class AvailabilityZoneValidator(Validator):
+
+    def _validate(self, login_node_subnet_id, head_node_subnet_id):
+        if AWSApi.instance().ec2.get_subnet_avail_zone(login_node_subnet_id) != \
+                AWSApi.instance().ec2.get_subnet_avail_zone(head_node_subnet_id):
+            self._add_failure(
+                "LoginNode Networking SubnetId must be in the same availability zone as the HeadNode.",
```

Comment:
Messaging will need rewording but let's go for a tentative message: "Login nodes and head node must be in the same availability zone".

Reply:
Done!